ECS compatibility issue: BasicView to MultiView type mismatch in zig_ecs adapter (v0.26.0)

## Description

There is a type compatibility issue between labelle-engine v0.26.0 and the zig_ecs backend that prevents projects from compiling when using custom components with `onAdd`/`onRemove` callbacks.

## Environment

- **labelle-engine version**: 0.26.0
- **Zig version**: 0.15.2
- **ECS backend**: zig_ecs
- **Platform**: macOS (Apple M1 Max)
- **labelle-engine commit**: 54bea21 (main branch, latest)

## Error Details

When attempting to compile a project with custom components (Storage, Workstation) that have `onAdd`/`onRemove` callbacks, the build fails with the following errors:

```
/Users/alexandrecalvao/.cache/zig/p/labelle_engine-0.26.0-rhO5vhCACAA3KPaddaRqcsViNKEAhamMkAsBGUC66Xy-/src/ecs/zig_ecs_adapter.zig:170:31: error: expected type 'ecs.views.MultiView(.{ components.workstation.Workstation },.{})', found 'ecs.views.BasicView(components.workstation.Workstation)'
        return self.inner.view(includes, .{});
               ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
/Users/alexandrecalvao/.cache/zig/p/entt-1.0.0-qJPtbNLVAgDPXaUbyRSsBTK75Cr1WujUA92kToJugWry/src/ecs/views.zig:13:12: note: struct declared here
    return struct {
           ^~~~~~
/Users/alexandrecalvao/.cache/zig/p/entt-1.0.0-qJPtbNLVAgDPXaUbyRSsBTK75Cr1WujUA92kToJugWry/src/ecs/views.zig:80:12: note: struct declared here
    return struct {
           ^~~~~~
/Users/alexandrecalvao/.cache/zig/p/labelle_engine-0.26.0-rhO5vhCACAA3KPaddaRqcsViNKEAhamMkAsBGUC66Xy-/src/ecs/zig_ecs_adapter.zig:169:79: note: function return type declared here
    pub fn view(self: *Registry, comptime includes: anytype) zig_ecs.MultiView(includes, .{}) {
                                                             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
```

Additionally, there's a formatNumber error:

```
/Users/alexandrecalvao/.zvm/0.15.2/lib/std/Io/Writer.zig:1048:43: error: no field or member function named 'formatNumber' in 'ecs.entity.EntityClass(.{ .index_bits = 20, .version_bits = 12 })'
                .@"struct" => return value.formatNumber(w, options.toNumber(.decimal, .lower)),
                                     ~~~~~^~~~~~~~~~~~~
/Users/alexandrecalvao/.cache/zig/p/entt-1.0.0-qJPtbNLVAgDPXaUbyRSsBTK75Cr1WujUA92kToJugWry/src/ecs/entity.zig:18:19: note: struct declared here
    return packed struct(EntityBackingInt) {
           ~~~~~~~^~~~~~
```

## Root Cause

The issue appears to be in `src/ecs/zig_ecs_adapter.zig` line 169-170:

```zig
pub fn view(self: *Registry, comptime includes: anytype) zig_ecs.MultiView(includes, .{}) {
    return self.inner.view(includes, .{});
}
```

When `includes` contains a single component type (e.g., `.{Workstation}` or `.{Storage}`), the underlying `zig_ecs.Registry.view()` returns a `BasicView(ComponentType)`, but the adapter's return type signature expects a `MultiView(includes, .{})`.

This type mismatch occurs specifically when:
1. Custom components have `onAdd`/`onRemove` callbacks
2. Scripts or internal engine code query for entities with these components using `registry.view(.{ComponentType})`

## Reproduction Steps

1. Create a new labelle-engine 0.26.0 project with zig_ecs backend
2. Create a custom component with an onAdd callback:

```zig
// components/storage.zig
const std = @import("std");
const engine = @import("labelle-engine");

pub const Storage = struct {
    storage_type: enum { eis, iis, ios, eos },
    
    pub fn onAdd(payload: engine.ComponentPayload) void {
        std.log.warn("[Storage.onAdd] Entity {d} added", .{payload.entity_id});
    }
    
    pub fn onRemove(payload: engine.ComponentPayload) void {
        std.log.warn("[Storage.onRemove] Entity {d} removed", .{payload.entity_id});
    }
};
```

3. Register the component in components/items.zig
4. Add entities with this component in a scene file
5. Create a script that queries for these components:

```zig
pub fn update(game: *Game, scene: *Scene, dt: f32) void {
    const registry = game.getRegistry();
    var view = registry.view(.{Storage});  // This triggers the error
    var iter = view.entityIterator();
    // ...
}
```

6. Run `labelle run`

## Expected Behavior

The project should compile successfully, and components with onAdd/onRemove callbacks should work seamlessly with ECS queries.

## Actual Behavior

Compilation fails with type mismatch errors in zig_ecs_adapter.zig.

## Workaround

Reverting to labelle-engine v0.19.0 resolves the issue:

```bash
cd path/to/labelle-engine
git checkout v0.19.0
zig build install --prefix ~/.local
cd path/to/project
labelle update
```

The project compiles and runs successfully with v0.19.0.

## Additional Context

- The issue seems to have been introduced between v0.19.0 and v0.26.0
- The zig_ecs dependency (entt-1.0.0) differentiates between `BasicView` (for single component queries) and `MultiView` (for multiple components)
- The adapter's return type always expects `MultiView`, but zig_ecs returns `BasicView` for single-component queries
- This might be related to changes in how component callbacks are registered or how views are created internally

## Possible Solutions

1. **Type-based conditional return**: Make the adapter return different view types based on the number of components in `includes`
2. **Normalize to MultiView**: Ensure the underlying zig_ecs call always returns a MultiView, even for single components
3. **Wrapper type**: Create a unified view wrapper that can handle both BasicView and MultiView
4. **Update zig_ecs dependency**: Check if a newer version of the zig_ecs backend resolves this compatibility issue

## Impact

This is a **blocking issue** for any project using:
- labelle-engine v0.26.0
- zig_ecs backend  
- Custom components with lifecycle callbacks
- Scripts that query for entities with these components

Projects cannot upgrade from v0.19.0 to v0.26.0 without removing their custom component callbacks or ECS queries.

--- a/src/ecs/zig_ecs_adapter.zig
+++ b/src/ecs/zig_ecs_adapter.zig
@@ -167,6 +167,16 @@ pub const Registry = struct {
     /// Determine the view type based on the number of components
     /// Single component -> BasicView (optimized), multiple -> MultiView
     fn ViewType(comptime includes: anytype) type {
+        comptime {
+            const T = @TypeOf(includes);
+            const ti = @typeInfo(T);
+            if (ti != .@"struct" or !ti.@"struct".is_tuple) {
+                @compileError("view() expects a tuple of types, e.g. '.{MyComponent}'");
+            }
+            if (includes.len == 0) {
+                @compileError("view() requires at least one component type; empty tuples are not supported");
+            }
+        }
         if (includes.len == 1) return zig_ecs.BasicView(includes[0]);
         return zig_ecs.MultiView(includes, .{});
     }

--- a/test/component_callbacks_test.zig
+++ b/test/component_callbacks_test.zig
@@ -569,14 +569,21 @@ pub const VIEW_WITH_CALLBACKS = struct {
         // Query using single-component view - this was causing the type mismatch
         var view = registry.view(.{TestHealth});
         var count: u32 = 0;
+        var found_100: bool = false;
+        var found_50: bool = false;
         var iter = view.entityIterator();
         while (iter.next()) |entity| {
             const health = registry.tryGet(TestHealth, entity);
             try std.testing.expect(health != null);
+            // Verify we can access the actual component values
+            if (health.?.amount == 100) found_100 = true;
+            if (health.?.amount == 50) found_50 = true;
             count += 1;
         }
 
         try expect.equal(count, 2);
+        try expect.toBeTrue(found_100);
+        try expect.toBeTrue(found_50);
     }
 
     test "multi-component view works with components that have callbacks" {
@@ -599,7 +606,14 @@ pub const VIEW_WITH_CALLBACKS = struct {
         var view = registry.view(.{ TestHealth, TestMana });
         var count: u32 = 0;
         var iter = view.entityIterator();
-        while (iter.next()) |_| {
+        while (iter.next()) |e| {
+            // Verify we can access both component values correctly
+            const health = registry.tryGet(TestHealth, e);
+            const mana = registry.tryGet(TestMana, e);
+            try std.testing.expect(health != null);
+            try std.testing.expect(mana != null);
+            try expect.equal(health.?.amount, 100);
+            try expect.equal(mana.?.current, 50);
             count += 1;
         }
 

